src/models/model.ts

```diff
@@ -0,0 +1,44 @@
+import { Attribute } from './attributes';
+
+export type SaveCallback = (error: Error | null, result?: any) => void;
+
+export default class Model {
+  static attributes: { [key: string]: Attribute };
+
+  constructor(json?: any) {
+    if (json) {
+      this.fromJSON(json);
+    }
+  }
+
+  attributes(): { [key: string]: Attribute } {
```

Comment:
Why is this needed?

Reply:
I don't think it's _needed_ I think it's just there to make the code look cleaner by inserting `this.attributes()` instead of `(this.constructor as any).attributes`.